ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -84,6 +92,14 @@ public LogicalSchema(
     this.metaFields = ImmutableList.copyOf(requireNonNull(metaFields, "metaFields"));
     this.keyFields = ImmutableList.copyOf(requireNonNull(keyFields, "keyFields"));
     this.valueFields = ImmutableList.copyOf(requireNonNull(valueFields, "valueFields"));
+
+    final BiMap<Field, String> fieldNameBiMap = HashBiMap.create();
```

Comment:
This does not belong in this class.  The _logical_ schema is not the place to store the internal field names used by the KStream implementation.